tests/rust-integration-tests/integration_test/src/tests/hooks/hooks_test.rs

```diff
@@ -0,0 +1,87 @@
+use anyhow::anyhow;
+use oci_spec::runtime::{Hook, HookBuilder, HooksBuilder, ProcessBuilder, Spec, SpecBuilder};
+use std::{fs::File, io::Read};
+use test_framework::{Test, TestGroup, TestResult};
+
+use crate::utils::{
+    create_container, delete_container, generate_uuid, prepare_bundle, set_config,
+    test_utils::start_container,
+};
+
+fn write_log_hook(content: &str) -> Hook {
+    HookBuilder::default()
+        .path("/bin/sh")
+        .args(vec![
+            "sh".to_string(),
+            "-c".to_string(),
+            format!("echo '{}' >> {}", content, "output"),
+        ])
+        .build()
+        .expect("could not build hook")
+}
+
+fn get_spec() -> Spec {
+    SpecBuilder::default()
+        .process(
+            ProcessBuilder::default()
+                .args(vec!["true".to_string()])
+                .build()
+                .unwrap(),
+        )
+        .hooks(
+            HooksBuilder::default()
+                .prestart(vec![
+                    write_log_hook("pre-start1 called"),
```

Comment:
Why do we need to call twice per-hook?

Reply:
It's ported from here:
https://github.com/opencontainers/runtime-tools/blob/0105384f68e16803891d0a17d9067b1def6a2778/validation/hooks/hooks.go#L32-L67
I was having the same question. I guess calling twice is to ensure the implementation actually executes hooks in the given order. (hook 1 then hook 2 in sequence)